### PR TITLE
Spell Delta table consistently

### PR DESCRIPTION
**Pull request by stbof**
Fixes a couple of "delta" and changes Delta Lake table to Delta table.

### CODE REVIEW

File: PROTOCOL.md [hunk 1]
```diff
@@ -253,7 +253,7 @@ The following is an example `remove` action.
 
 ### Transaction Identifiers
 Incremental processing systems (e.g., streaming systems) that track progress using their own application-specific versions need to record what progress has been made, in order to avoid duplicating data in the face of failures and retries during a write.
-Transaction identifiers allow this information to be recorded atomically in the transaction log of a delta table along with the other actions that modify the contents of the table.
+Transaction identifiers allow this information to be recorded atomically in the transaction log of a Delta table along with the other actions that modify the contents of the table.
```
Comment (mukulmurthy) [inline PROTOCOL.md]: This change is valid, but the other changes in this file refer to delta (like change) files (which contain incremental information about commits to the Delta table), not Delta table files. I think those should stay lowercase.

Comment (stbof) [reply PROTOCOL.md]: Fixed.

**Comment by mukulmurthy** [comment]
There's a few other places in the repo where we say "Delta Lake" instead of just "Delta". Mind if I fix those too when merging this?

**Comment by tdas** [comment]
@mukulmurthy I think the policy and @stbof and I agreed on yesterday was that any references to a specific table or the DeltaTable class/instance should be referred in the APIs docs as "Delta table" to keep it consistent with the DeltaTable class. But any reference to the project should still be called Delta Lake. Update accordingly.

**Comment by mukulmurthy** [comment]
We have lots of cases where we say just "Delta Lake" but actually mean "Delta". The easiest fix is to just remove the Lake part when it's being used to only one table. There's a bunch of examples in https://github.com/delta-io/delta/blob/master/src/main/scala/org/apache/spark/sql/delta/DeltaErrors.scala#L197.

**Comment by databricks-cla-assistant** [comment]
[CLA assistant check](https://cla.dev.databricks.com/delta-io/delta?pullRequest=207) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.databricks.com/delta-io/delta?pullRequest=207) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dev.databricks.com/check/delta-io/delta?pullRequest=207) it.</sub>

**Comment by zsxwing** [comment]
Closing the stale PR

**Pull request closed without merging**